aptos-move/framework/aptos-stdlib/sources/cryptography/bulletproofs.move

```diff
@@ -0,0 +1,253 @@
+    // Range proof for A_COMM using domain-separation tag in A_DST, and MAX_RANGE_BITS
+    #[test_only]
+    const A_RANGE_PROOF_PEDERSEN: vector<u8> = x"d8d422d3fb9511d1942b78e3ec1a8c82fe1c01a0a690c55a4761e7e825633a753cca816667d2cbb716fe04a9c199cad748c2d4e59de4ed04fedf5f04f4341a74ae75b63c1997fd65d5fb3a8c03ad8771abe2c0a4f65d19496c11d948d6809503eac4d996f2c6be4e64ebe2df31102c96f106695bdf489dc9290c93b4d4b5411fb6298d0c33afa57e2e1948c38ef567268a661e7b1c099272e29591e717930a06a2c6e0e2d56aedea3078fd59334634f1a4543069865409eba074278f191039083102a9a0621791a9be09212a847e22061e083d7a712b05bca7274b25e4cb1201c679c4957f0842d7661fa1d3f5456a651e89112628b456026f8ad3a7abeaba3fec8031ec8b0392c0aa6c96205f7b21b0c2d6b5d064bd5bd1a1d91c41625d910688fa0dca35ec0f0e31a45792f8d6a330be970a22e1e0773111a083de893c89419ee7de97295978de90bcdf873a2826746809e64f9143417dbed09fa1c124e673febfed65c137cc45fabda963c96b64645802d1440cba5e58717e539f55f3321ab0c0f60410fba70070c5db500fee874265a343a2a59773fd150bcae09321a5166062e176e2e76bef0e3dd1a9250bcb7f4c971c10f0b24eb2a94e009b72c1fc21ee4267881e27b4edba8bed627ddf37e0c53cd425bc279d0c50d154d136503e54882e9541820d6394bd52ca2b438fd8c517f186fec0649c4846c4e43ce845d80e503dee157ce55392188039a7efc78719107ab989db8d9363b9dfc1946f01a84dbca5e742ed5f30b07ac61cf17ce2cf2c6a49d799ed3968a63a3ccb90d9a0e50960d959f17f202dd5cf0f2c375a8a702e063d339e48c0227e7cf710157f63f13136d8c3076c672ea2c1028fc1825366a145a4311de6c2cc46d3144ae3d2bc5808819b9817be3fce1664ecb60f74733e75e97ca8e567d1b81bdd4c56c7a340ba00";
+
+    #[test(fx = @std)]
+    #[expected_failure(abort_code = 0x010003, location = Self)]
+    fun test_unsupported_ranges(fx: signer) {
+        features::change_feature_flags(&fx, vector[ features::get_bulletproofs_feature() ], vector[]);
+
+        let comm = ristretto255::new_point_from_bytes(A_COMM);
+        let comm = std::option::extract(&mut comm);
+        let comm = pedersen::commitment_from_point(comm);
+
+        assert!(verify_range_proof_pedersen(
+            &comm,
+            &range_proof_from_bytes(A_RANGE_PROOF_PEDERSEN), 10, A_DST), 1);
+    }
+
+    #[test(fx = @std)]
+    fun test_prover(fx: signer) {
+        features::change_feature_flags(&fx, vector[ features::get_bulletproofs_feature() ], vector[]);
+
+        let v = ristretto255::new_scalar_from_u64(59);
+        let r = ristretto255::new_scalar_from_bytes(A_BLINDER);
+        let r = std::option::extract(&mut r);
+        let num_bits = 8;
+
+        let (proof, comm) = prove_range_pedersen(&v, &r, num_bits, A_DST);
+
+        assert!(verify_range_proof_pedersen(&comm, &proof, 64, A_DST) == false, 1);
+        assert!(verify_range_proof_pedersen(&comm, &proof, 32, A_DST) == false, 1);
+        assert!(verify_range_proof_pedersen(&comm, &proof, 16, A_DST) == false, 1);
+        assert!(verify_range_proof_pedersen(&comm, &proof, num_bits, A_DST), 1);
+    }
+
+    #[test(fx = @std)]
+    #[expected_failure(abort_code = 0x010001, location = Self)]
+    fun test_empty_range_proof(fx: signer) {
```

Comment:
what does `fx` stand for? sorry i should have asked 4 months ago.

Reply:
Framework :)